dogstatsd-client/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+                    do_send(client.count_with_tags(metric.as_ref(), value), &tags)
+                }
+                DogStatsDActionOwned::Distribution(metric, value, tags) => {
+                    do_send(client.distribution_with_tags(metric.as_ref(), value), &tags)
+                }
+                DogStatsDActionOwned::Gauge(metric, value, tags) => {
+                    do_send(client.gauge_with_tags(metric.as_ref(), value), &tags)
+                }
+                DogStatsDActionOwned::Histogram(metric, value, tags) => {
+                    do_send(client.histogram_with_tags(metric.as_ref(), value), &tags)
+                }
+                DogStatsDActionOwned::Set(metric, value, tags) => {
+                    do_send(client.set_with_tags(metric.as_ref(), value), &tags)
+                }
+            } {
+                error!("Error while sending metric: {}", err);
+            }
+        }
+    }
+
+    /// Send a vector of DogStatsDAction, this is the same as `send_owned` except it only borrows
+    /// the provided values.See the docs for DogStatsDActionOwned for details.
+    pub fn send<'a, T: AsRef<str>, V: IntoIterator<Item = &'a Tag>>(
+        &self,
+        actions: Vec<DogStatsDAction<'a, T, V>>,
+    ) {
+        let client = &self.client;
+
+        for action in actions {
+            if let Err(err) = match action {
+                DogStatsDAction::Count(metric, value, tags) => {
+                    let metric_builder = client.count_with_tags(metric.as_ref(), value);
+                    do_send(metric_builder, tags)
+                }
+                DogStatsDAction::Distribution(metric, value, tags) => {
+                    do_send(client.distribution_with_tags(metric.as_ref(), value), tags)
+                }
+                DogStatsDAction::Gauge(metric, value, tags) => {
+                    do_send(client.gauge_with_tags(metric.as_ref(), value), tags)
+                }
+                DogStatsDAction::Histogram(metric, value, tags) => {
+                    do_send(client.histogram_with_tags(metric.as_ref(), value), tags)
+                }
+                DogStatsDAction::Set(metric, value, tags) => {
+                    do_send(client.set_with_tags(metric.as_ref(), value), tags)
+                }
+            } {
+                error!("Error while sending metric: {}", err);
+            }
+        }
+    }
+}
+
+fn do_send<'m, 't, T, V: IntoIterator<Item = &'t Tag>>(
+    mut builder: MetricBuilder<'m, '_, T>,
+    tags: V,
+) -> anyhow::Result<()>
+where
+    T: Metric + From<String>,
+    't: 'm,
+{
+    let mut tags_iter = tags.into_iter();
+    let mut tag_opt = tags_iter.next();
+    while tag_opt.is_some() {
+        builder = builder.with_tag_value(tag_opt.unwrap().as_ref());
+        tag_opt = tags_iter.next();
+    }
+    builder.try_send()?;
+    Ok(())
+}
+
+fn create_client(endpoint: &Endpoint) -> anyhow::Result<StatsdClient> {
```

Comment:
Nit: as the functions has multiple points for returning an error and these errors can bubble up in the stack frame I would suggest to add more context creating some explicit errors with `anyhow` so it's easier to debug which method failed.

Reply:
good call out, I've just added additional `anyhow` in the places it wasn't used before!